subsys/bluetooth/host/buf.c

```diff
@@ -147,3 +151,62 @@ struct net_buf_pool *bt_buf_get_num_complete_pool(void)
 }
 #endif /* CONFIG_BT_CONN || CONFIG_BT_ISO */
 #endif /* ZTEST_UNITTEST */
+
+struct net_buf *bt_buf_make_view(struct net_buf *view,
+				 struct net_buf *parent,
+				 size_t len,
+				 struct bt_buf_view_meta *meta)
+{
+	__ASSERT_NO_MSG(len);
+	__ASSERT_NO_MSG(view);
```

Comment:
Implicit casts to `bool` does not give a warning with the current compiler arguments, but it might be added in the future.

```suggestion
	__ASSERT_NO_MSG(len != NULL);
	__ASSERT_NO_MSG(view != NULL);
```

Reply:
What result type has the equality operator in C?

Reply:
https://www.phind.com/search?cache=qs1npuw15vr0ftyak6txd1n3

Reply:
Until C23 it will technically be an `int` yes :D 

There is still a difference though

```
__ASSERT_NO_MSG(view); /* assets if view is non-zero */
__ASSERT_NO_MSG(view == true); /* assets if view is 1 */
__ASSERT_NO_MSG(view != NULL); /* assets if view is non-NULL */
__ASSERT_NO_MSG((view != NULL) == true); /* assets if view is non-NULL */
```

Besides making the code future proof, `view != NULL` is semantically more correct as it can be compared to the `bool` macros from `stdbool.h`.

Reply:
C23 is still draft, but I do not see result type semantic changed (and likely will never be). https://web.archive.org/web/20230402172459/https://www.open-std.org/jtc1/sc22/wg14/www/docs/n3096.pdf

Future yes, https://www.phind.com/search?cache=wvtqmebws5vq5o21ck7ms4bz

Reply:
You are correct. The return value of `==` will actually remain an `int`, and the new boolean types will seemingly be backwards compatible: 

> 6.3.1.2 Boolean type
> When any scalar value is converted to bool, the result is false if the value is a zero (for arithmetic types), null (for pointer types), or the scalar has type nullptr_t; otherwise, the result is true.

So please ignore my above references to C23; clearly I misread something earlier. 

My position on the semantics still remain though :)

Reply:
Not a problem anyways, we know everyone just disables asserts in prod 🫠